tools-local/tasks/GenerateFileVersionProps.cs

```diff
@@ -58,15 +58,21 @@ public override bool Execute()
 
                 if (fileVersions.TryGetValue(fileName, out existing))
                 {
-                    if (current.AssemblyVersion != null &&
-                        existing.AssemblyVersion != null &&
-                        current.AssemblyVersion != existing.AssemblyVersion)
+                    if (current.AssemblyVersion != null)
                     {
-                        if (current.AssemblyVersion > existing.AssemblyVersion)
+                        if (existing.AssemblyVersion == null)
```

Comment:
Curious how we had any assembly version that was null once and not null the second time.  Was the first time a cross-gen image for a different platform?  If so, maybe we should switch our assembly version reading to always use SRM (see https://github.com/dotnet/sdk/issues/1502)

Reply:
When I debugged it on my machine, it is because the first file encountered doesn't exist.

`F:\git\core-setup\Bin\win-x64.Debug\crossgen\Microsoft.NETCore.App\linux-x64\runtimes\linux-x64\lib\netcoreapp2.1\System.Runtime.CompilerServices.Unsafe.dll`

Notice that it is `crossgen\Microsoft.NETCore.App\linux-x64` and I'm on a Windows machine.

https://github.com/dotnet/core-setup/blob/a0b82c4ec4a5186d1ecf611503bd075a308be7c6/tools-local/tasks/GenerateFileVersionProps.cs#L123-L137

Notice that if the file doesn't exist, it tries to get the AssemblyVersion from the metadata, which also doesn't exist.

Reply:
I see, that makes sense.  I bet we broke this when we started cross-gening.  Prior to that we'd actually restore all the RIDs when building out this manifest just to read the file info out of them.